src/components/Personalization/createApplyPropositions.js

```diff
@@ -12,61 +12,78 @@ governing permissions and limitations under the License.
 
 import composePersonalizationResultingObject from "./utils/composePersonalizationResultingObject";
 import isNonEmptyArray from "../../utils/isNonEmptyArray";
-import isEmptyObject from "../../utils/isEmptyObject";
 import isObject from "../../utils/isObject";
+import { DOM_ACTION, HTML_CONTENT_ITEM } from "./constants/schema";
+import PAGE_WIDE_SCOPE from "./constants/scope";
 
 export const EMPTY_PROPOSITIONS = { propositions: [] };
-export const HTML_CONTENT_ITEM = "html-content-item";
-export const DEFAULT_METADATA = {
-  selector: "head",
-  type: "appendHtml"
-};
+export const SUPPORTED_SCHEMAS = [DOM_ACTION, HTML_CONTENT_ITEM];
 
-export default ({ executeDecisions, showContainers }) => {
-  const updatePropositions = ({ proposition, metadataForScope }) => {
-    proposition.items.forEach(item => {
-      if (item.schema.includes(HTML_CONTENT_ITEM)) {
-        item.data.selector =
-          metadataForScope.selector || DEFAULT_METADATA.selector;
-        item.data.type = metadataForScope.actionType || DEFAULT_METADATA.type;
-      }
-    });
+export default ({ executeDecisions }) => {
+  const filterItemsPredicate = item => {
+    if (SUPPORTED_SCHEMAS.includes(item.schema)) {
+      return true;
+    }
+    return false;
+  };
+
+  const updatePropositionItems = ({ proposition, metadataForScope }) => {
+    proposition.items = proposition.items
+      .filter(filterItemsPredicate)
+      .map(item => {
+        if (item.schema === HTML_CONTENT_ITEM && isObject(metadataForScope)) {
+          item.data.selector = metadataForScope.selector;
+          item.data.type = metadataForScope.actionType;
```

Comment:
This mutates the original item. Inside the if you could use this to return a new object. Also you can simplify things if you just return the new items.

```
const updatePropositionItems = ({ items, metadataForScope }) => {
  return items
    .filter(filterItemsPredicate)
    .map(item => {
      if (item schema === HTML_CONTENT_ITEM && isObject(metadataForScope)) {
        return {
          ...item,
          data: {
            ...item.data,
            selector: metadataForScope.selector,
            type: metadataForScope.actionType
          }
        };
      }
      return item;
    });
};
```